rero_ils/modules/patrons/api.py

```diff
@@ -124,15 +125,19 @@ def validate(self, **kwargs):
                     not_required={}
                 ) or True
             if self.is_librarian:
-                validation_message = pids_exists_in_data(
-                    info='{pid_type} ({pid})'.format(
-                        pid_type=self.provider.pid_type,
-                        pid=self.pid
-                    ),
-                    data=self,
-                    required={'lib': 'library'},
-                    not_required={}
-                ) or True
+                libraries = self.get('libraries')
+                if not libraries:
+                    validation_message = 'Missing libraries'
+                for library in self.get('libraries', []):
+                    if '$ref' in library:
+                        rec = extracted_data_from_ref(
+                            library['$ref'], 'record')
+                        if not rec:
+                            validation_message =\
+                                'Library {ref} doesn\'t exist.'.format(
+                                    ref=library['$ref']
+                                )
+                            break
```

Comment:
```
for library_pid in self.library_pids:
    library = Library.get_record_by_pid(library_pid)
    if library is None:
        validation_message = ...
        break
```
As we have access to `self` we can call self property ;-)

Reply:
done